pkg/utils/mco_managedcluster.go

```diff
@@ -31,3 +39,94 @@ func UpdateObservabilityFromManagedCluster(opt TestOptions, enableObservability
 	}
 	return nil
 }
+
+func ListManagedClusters(opt TestOptions) ([]string, error) {
+	clientDynamic := GetKubeClientDynamic(opt, true)
+	objs, err := clientDynamic.Resource(NewOCMManagedClustersGVR()).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	clusterNames := []string{}
+	for _, obj := range objs.Items {
+		metadata := obj.Object["metadata"].(map[string]interface{})
+		name := metadata["name"].(string)
+		labels := metadata["labels"].(map[string]interface{})
+		if labels != nil {
+			obsControllerStr := ""
+			if obsController, ok := labels["feature.open-cluster-management.io/addon-observability-controller"]; ok {
+				obsControllerStr = obsController.(string)
+			}
+			if obsControllerStr != "unreachable" {
+				clusterNames = append(clusterNames, name)
+			}
+		}
+	}
+
+	if len(clusterNames) == 0 {
+		return clusterNames, fmt.Errorf("no managedcluster found")
+	}
+
+	return clusterNames, nil
+}
```

Comment:
Looks like you don't use this function. better to remove this, because the label `"feature.open-cluster-management.io/addon-observability-controller"` in that function is only available in new version.